client/components/domains/register-domain-step/index.jsx

```diff
@@ -168,17 +169,14 @@ class RegisterDomainStep extends React.Component {
 		if ( props.initialState ) {
 			this.state = { ...this.state, ...props.initialState };
 
-			if (
-				this.state.lastSurveyVertical &&
-				this.state.lastSurveyVertical !== props.surveyVertical
-			) {
+			if ( this.state.lastVerticalId && this.state.lastVerticalId !== props.verticalId ) {
 				this.state.loadingResults = true;
 
 				if ( props.includeWordPressDotCom || props.includeDotBlogSubdomain ) {
 					this.state.loadingSubdomainResults = true;
 				}
 
-				delete this.state.lastSurveyVertical;
```

Comment:
super unrelated to this PR but is it really directly modifying `this.state`? :o

Reply:
@adamziel I know this looks an anti-pattern. But this is the only way, as far as I know, to reset a state variable in the constructor as you can't use `setState` here. :)